### DISCUSSION TITLE
Discussion on potential need for zsync replacement (or other steps for the future)

### DISCUSSION THREAD

**Discussion by AyoungDukie** [quickemu-project/quickemu#987] [Development chat]
At the start of 2024, the zsync [project's hosting went down](http://web.archive.org/web/20240101202329/http://zsync.moria.org.uk/) and their last mailing list activity seems to have been in 2016 (though it seems from archive.org that it was up the prior week). It has since been brought back up. Their source is still up on Github to allow compiling from source (and there are already built binaries in repos, etc), but the episode made me wonder if an alternative could be good to evaluate for the future? Or perhaps it may be worth forking to bring into active maintenance?

Some similar tools I've seen discussions around
- [zsync-curl](https://github.com/probonopd/zsync-curl) (still last updated 3 years ago)
- [zsync2](https://github.com/AppImageCommunity/zsync2) (seems to still be WIP and may not be fully stable)
- [casync](https://github.com/systemd/casync) (seems to address similar uses, and is actively maintained by the systemd team)
- [desync](https://github.com/folbricht/desync) (a casync implementation in Go)

FWIW, it does look like folks looked into utilizing casync for Nix purposes (though this effort was rolled into [Tvix](https://cs.tvl.fyi/depot/-/blob/tvix/README.md).

~~Otherwise, if anyone knows who to contact, perhaps the zsync hosting site could be renewed?~~

**Comment by TuxVinyards** [comment]
The zsync routine in qqX is pretty polished now.  

Unlike vanilla quickemu, it tells you how many hours since you last updated the daily-live. and gives you download progress reports etc  

Can't say that I really want to have to redo this when it's working well.

The link given in Synaptic works fine for me.  <http://zsync.moria.org.uk/downloads>

The source code is quite old but seems pretty good still.   Not any critical issues. <https://github.com/cph6/zsync>

**Comment by AyoungDukie** [comment]
It looks like the site has been brought back up, and I've updated my post accordingly!

Also fair point on the potential rework requirements!

**Comment by TuxVinyards** [comment]
The other thing to note, is that zsync is also used by Ubuntu at their end too.

Check out <https://cdimage.ubuntu.com/daily-live/current/> for example ...

**Comment by flexiondotorg** [comment]
I think zsync support needs to remain for as long as projects are publishing `.zsync` resources, but I'd be up for including support for compatible alternatives with a fallback to `zsync`.

**Comment by Dal1ster** [comment]
i do think that there should be an alternative to zsync since for example, on Fedora 40 trying to get zsync from the setup instructions put that it doesent exist and trying to self compile it/install it causes it to fail to build, rendering it impossible to run quickemu

**Reply by AyoungDukie** [reply]
I was able to work around this on my own COPR build by putting it as a recommended package instead of required. For quickemu, it hasn't caused me direct issues in my limited use case, but allowed me the option to find/use it from a different repo for other use (e.g. in qqX).

May be worth trying?

Edit: For reference, my Spec file is here:
https://github.com/AyoungDukie/Personal/blob/main/quickemu.spec

**Reply by AyoungDukie** [reply]
If I recall, I think I found it in RPM Sphere for installing the package, though I'm sure one could set up a COPR and build the RPM as an alternative:
https://pkgs.org/download/zsync
![image](https://github.com/user-attachments/assets/fcafe56e-c8b6-4f40-9cc9-077d2309ad0d)

Their homepage, for reference:
https://rpmsphere.github.io/
